src/panel/App.tsx

```diff
@@ -12,27 +13,40 @@ import {
   DevtoolsProvider,
   EventsProvider,
   RequestProvider,
-  ExplorerContextProvider
+  ExplorerContextProvider,
+  useDevtoolsContext
 } from "./context";
 
 export const App = () => {
   return (
-    <DevtoolsProvider>
-      <ThemeProvider theme={theme}>
-        <HashRouter>
-          <EventsProvider>
-            <Route path="/events" component={Events} />
-          </EventsProvider>
-          <RequestProvider>
-            <Route path="/request" component={Request} />
-          </RequestProvider>
-          <ExplorerContextProvider>
-            <Route path="/explorer" exact component={Explorer} />
-          </ExplorerContextProvider>
-          <Route path="/" exact component={() => <Redirect to="/explorer" />} />
-          <Navigation />
-        </HashRouter>
-      </ThemeProvider>
-    </DevtoolsProvider>
+    <ThemeProvider theme={theme}>
+      <DevtoolsProvider>
+        <AppRoutes />
+      </DevtoolsProvider>
+    </ThemeProvider>
+  );
+};
+
+export const AppRoutes: FC = () => {
```

Comment:
I personally prefer having one component in one file, don't know how you perceive this. (more to spark a conversation about this)

Reply:
> I personally prefer having one component in one file

For the most part we've been taking this approach. 

In this case I'm not sure there's much value in breaking it down into separate files - there's only two components and they would be (and were) one if it wasn't for the use of context/hooks.